src/main/webapp/script.js

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+window.addEventListener('load', () => {
+  let selected = [];  // Courses selected by the user
+  let courses = [];   // List with all courses
+
+  getOptions();
+
+  document.getElementById('add-selected').addEventListener('click', function() {
+    addToSelected();
+  });
+
+  /**
+   * Gets courses from /courselist servlet to populate dropdown list
+   */
+  async function getOptions() {
+    const response = await fetch('/courses');
+    courseList = await response.json();
+    const courseContainer = document.getElementById('courses');
+    courseContainer.innerHTML = '';
+    addOption('Select a Course', courseContainer, /*shouldSetValue=*/ false);
+    coursesDetailed = JSON.parse(courseList.courses_detailed);
```

Comment:
why do you need to `JSON.parse`? This should already be de-stringified since you called `response.json()` above. If you actually need to do this for it to work, then you need to fix the response from the servlet itself so you aren't double-encoding this in json.

Reply:
It decodes the response in JSON, but the inner JSON objects are not decoded -- when I didn't include this, it treated it as a string.

Reply:
This is pretty hard evidence that you have a bug in the servlet if you need to decode twice.